qiskit_qec/circuits/repetition_code.py

```diff
@@ -362,3 +368,703 @@ def partition_outcomes(
         final_outcomes = [int(c) for c in finals]
 
         return x_gauge_outcomes, z_gauge_outcomes, final_outcomes
+
+
+def add_edge(graph, pair, edge=None):
+    """
+    Adds an edge correspoding to the given pair of nodes to the given graph,
+    adding also the nodes themselves if not already present.
+    """
+    ns = []
+    for node in pair:
+        if node not in graph.nodes():
+            ns.append(graph.add_node(node))
+        else:
+            ns.append(graph.nodes().index(node))
+    graph.add_edge(ns[0], ns[1], edge)
+    return ns
+
+
+class ArcCircuit:
```

Comment:
At some point we need to figure out common interface for this circuit classes. The suppose to share at least something :) 

Maybe something like this:
```python
class CodeCircuit:
    def circuit() -> QuantumCircuit:
        """Generate corresponding circuit to given code parameters."""
        raise NotImplementedError
```

and then in ArcCircuit you will do something like this
```python
class ArcCircuit(CodeCircuit):
    ...
    def circuit() -> QuantumCircuit:
        ...
        return self.transpile(...)
```

We at least introduce some type annotation, so for developers/users it will be easier to handle.

I would like to hear your thoughts on this maybe next week on a standup :)